Formula/tidb-server.rb

```diff
@@ -0,0 +1,75 @@
+class TidbServer < Formula
+  desc "TiDB is a MySQL compatible distributed database"
+  homepage "https://www.pingcap.com/en/"
+  url "https://github.com/pingcap/tidb.git",
+      :tag      => "v2.1.5",
+      :revision => "4dc309315c845ee1e1955754325bf2fc212e8f8d"
+
+  depends_on "go" => :build
+
+  def install
+    # Change the default datadir
+    inreplace "config/config.go", "/tmp/tidb", var/"tidb"
+    inreplace "tidb-server/main.go", "/tmp/tidb", var/"tidb"
```

Comment:
I'm a bit puzzled by this. If the project is designed to run with that datadir in a temporary directory, we shouldn't move it to a permanent directory. That completely changes the purpose.

(I won't comment on the practice of hardcoding paths in the source in the first place)

Reply:
I am not sure of the historical reason it defaults to `/tmp/tidb`, it might be because of the test-suite.  If I leave it in `/tmp`, then the data is lost constantly.

The MySQL package uses a [cmake flag to set the datadir](https://github.com/Homebrew/homebrew-core/blob/master/Formula/mysql.rb#L56).  I also [commented [](here](https://github.com/Homebrew/homebrew-core/pull/37634#issuecomment-470970594)) why I made this change, and did not use a config file instead.  Long term this could be fixed by the `tidb-server` accepting a default config file.  I can not guarantee if it is fixed though, that it will be backported to the 2.1 branch.

Reply:
Then I'm tempted to say let's ship it as built and configured by upstream. They can fix it if they want, but we shouldn't have to work around their design decisions.

Reply:
I work for upstream if that was clear :-) I will work on the fix [described here](https://github.com/pingcap/tidb/issues/9621) shortly, but I can't guarantee a backport to the stable release series. It could break user systems if they have a file in `/etc/tidb.toml` and perform a minor version upgrade.  For the next major version, not a problem.

Shipping as configured though is not an expected behavior for users.

Reply:
> Shipping as configured though is not an expected behavior for users

Puzzling. Is each distributor supposed to write its own configuration file, or modify the paths in the source? How are Debian and other linux distros doing?

Reply:
TiDB is not in Debian or other Linux distros.

I believe there are arch personal packages (I am not sure how that system works).  You can see how it works in [dbdeployer here](https://github.com/datacharmer/dbdeployer/blob/master/sandbox/tidb_templates.go#L44-L45).

To me this is the combination of two problems: the default, and me trying to make it work without a configuration file so it will work more elegantly on linuxbrew (which does not support services, so I want to make it work for starting without specifying a config file).